src/mailkit.cs

```diff
@@ -0,0 +1,423 @@
+
+		[Static]
+		[Export ("warningWithLocalizedDescription:")]
+		MEAddressAnnotation AnnotateWarning (string localizedDescription);
+
+		[Static]
+		[Export ("successWithLocalizedDescription:")]
+		MEAddressAnnotation AnnotateSuccess (string localizedDescription);
+	}
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Advice ("'MEComposeSession' is not available in UIKit on macOS.")]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface MEComposeSession : NSSecureCoding
+	{
+		[Export ("sessionID", ArgumentSemantic.Strong)]
+		NSUuid SessionID { get; }
+
+		[Export ("mailMessage", ArgumentSemantic.Strong)]
+		MEMessage MailMessage { get; }
+
+		[Export ("reloadSession")]
+		void ReloadSession ();
+	}
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Advice ("'MEDecodedMessage' is not available in UIKit on macOS.")]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface MEDecodedMessage : NSSecureCoding
+	{
+		[Export ("initWithData:securityInformation:")]
+		IntPtr Constructor ([NullAllowed] NSData rawData, MEMessageSecurityInformation securityInformation);
+
+		[NullAllowed, Export ("rawData", ArgumentSemantic.Copy)]
+		NSData RawData { get; }
+
+		[Export ("securityInformation", ArgumentSemantic.Strong)]
+		MEMessageSecurityInformation SecurityInformation { get; }
+	}
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Advice ("'MEEncodedOutgoingMessage' is not available in UIKit on macOS.")]
+	[BaseType (typeof (NSObject))]
+	interface MEEncodedOutgoingMessage : NSSecureCoding
+	{
+		[Export ("initWithRawData:isSigned:isEncrypted:")]
+		IntPtr Constructor (NSData rawData, bool isSigned, bool isEncrypted);
+
+		[Export ("rawData", ArgumentSemantic.Copy)]
+		NSData RawData { get; }
+
+		[Export ("isSigned")]
+		bool IsSigned { get; }
+
+		[Export ("isEncrypted")]
+		bool IsEncrypted { get; }
+	}
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface MEEmailAddress : NSSecureCoding, NSCopying
+	{
+		[Export ("initWithRawString:")]
+		IntPtr Constructor (string rawString);
+
+		[Export ("rawString")]
+		string RawString { get; }
+
+		[NullAllowed, Export ("addressString")]
+		string AddressString { get; }
+	}
+
+	delegate void SessionHandler (NSDictionary<MEEmailAddress, MEAddressAnnotation> sessionDictionary);
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Protocol, Advice ("'MEComposeSessionHandler' is not available in UIKit on macOS.")]
+	[BaseType (typeof (NSObject))]
+	interface MEComposeSessionHandler
+	{
+		[Abstract]
+		[Export ("mailComposeSessionDidBegin:")]
+		void MailComposeSessionDidBegin (MEComposeSession session);
+
+		[Abstract]
+		[Export ("mailComposeSessionDidEnd:")]
+		void MailComposeSessionDidEnd (MEComposeSession session);
+
+		[Abstract]
+		[Export ("viewControllerForSession:")]
+		MEExtensionViewController GetViewController (MEComposeSession session);
+
+		[Async]
```

Comment:
can't put `Async` on protocols
conforming to this member just would not work at runtime
and since it's being called by the OS then it would not be helpful